### PR TITLE
add index option

### PR DESCRIPTION
**Pull request by hoyangtsai**

### CODE REVIEW

**Comment by justinmchase** [comment]
I think if you leave the default as `undefined` and then change the logic to be:

```
options = { index: @index }
```
Then pass in the options, if the default is undefined then it will behave as it does now by default if it is defined then it will be overridden. That should allow the tests to pass.

File: src/index.coffee [hunk 1]
```diff
@@ -123,9 +124,9 @@ class ConnectApp
       steps.unshift liveReload(@livereload)
     if typeof @root == "object"
       @root.forEach (path) ->
-        steps.push connect.static(path)
+        steps.push connect.static(path, {index: if @index is true then "index.html" else @index})
     else
-      steps.push connect.static(@root)
+      steps.push connect.static(@root, {index: if @index is true then "index.html" else @index})
```
Comment (justinmchase) [inline src/index.coffee]: Make an options variable above the `if` then pass that into both of these functions since they are identical.

Comment (hoyangtsai) [inline src/index.coffee]: I made the change like you said. Please check it out.

**Comment by justinmchase** [comment]
Published in release [4.2.0](https://github.com/AveVlad/gulp-connect/releases/tag/v4.2.0)

**Pull request merged**